src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -476,7 +476,10 @@ static inline ucs_status_t uct_dc_mlx5_iface_dci_get(uct_dc_mlx5_iface_t *iface,
         return UCS_OK;
     }
 
-    if (uct_dc_mlx5_iface_dci_can_alloc(iface)) {
+    /* Do not alloc dci if no CQ resources,
+     * otherwise this dci may never be released. */
+    if (uct_dc_mlx5_iface_dci_can_alloc(iface) &&
+        uct_rc_iface_has_tx_resources(&iface->super.super)) {
```

Comment:
q: do we need to check for TX pool entries availability?

Reply:
yes, because some `bcopy` operation may fail on taking the send desc, which would lead to the same situation.

Reply:
> yes, because some `bcopy` operation may fail on taking the send desc, which would lead to the same situation.

I see, thanks for the clarification